recipe/meta.yaml

```diff
@@ -291,41 +286,74 @@ outputs:
         - pip check
         - python -c "import torch; print(torch.__version__)"                       # [not (aarch64 and cuda_compiler_version != "None")]
         - python -c "import torch; assert torch.backends.mkldnn.m.is_available()"  # [x86 and cuda_compiler_version == "None"]
-        # Running tests on aarch64 + cuda is not supported until we move to Alma Linux 8
-        # https://github.com/conda-forge/pytorch-cpu-feedstock/pull/267#issuecomment-2387225526
         - python -c "import torch; torch.tensor(1).to('cpu').numpy(); print('numpy support enabled!!!')"
         # At conda-forge, we target versions of OSX that are too old for MPS support
         # But if users install a newer version of OSX, they will have MPS support
         # https://github.com/conda-forge/pytorch-cpu-feedstock/pull/123#issuecomment-1186355073
         # - python -c "import torch; assert torch.backends.mps.is_available()" # [osx]
+
+        # python-version-specific library (default location in SP_DIR symlinks back to this)
         - test -f $PREFIX/lib/libtorch_python${SHLIB_EXT}     # [unix]
 
+        # a reasonably safe subset of tests that should run under 15 minutes
+        # disable hypothesis because it randomly yields health check errors
+        {% set tests = " ".join([
+            "test/test_autograd.py",
+            "test/test_autograd_fallback.py",
+            "test/test_custom_ops.py",
+            "test/test_linalg.py",
+            "test/test_mkldnn.py",
+            "test/test_modules.py",
+            "test/test_nn.py",
+            "test/test_torch.py",
+            "test/test_xnnpack_integration.py",
+            "-m 'not hypothesis'",
+        ]) %}
+
+        {% set skips = "(TestTorch and test_print)" %}
+        # tolerance violation with openblas
+        {% set skips = skips ~ " or test_1_sized_with_0_strided_cpu_float32" %}         # [osx]
+        # timeouts and failures on aarch, see https://github.com/conda-forge/pytorch-cpu-feedstock/pull/298#issuecomment-2555888508
+        {% set skips = skips ~ " or test_pynode_destruction_deadlock" %}                # [aarch64]
+        {% set skips = skips ~ " or (TestLinalgCPU and test_cholesky_cpu_float32)" %}   # [aarch64]
+        {% set skips = skips ~ " or (TestLinalgCPU and test_pca_lowrank_cpu)" %}        # [aarch64]
+        {% set skips = skips ~ " or (TestLinalgCPU and test_svd_lowrank_cpu)" %}        # [aarch64]
+        {% set skips = skips ~ " or (TestMkldnnCPU and test_lstm_cpu)" %}               # [aarch64]
+        # dynamo does not support python 3.13
+        {% set skips = skips ~ " or (TestCustomOp and test_data_dependent_compile)" %}  # [py==313]
+        {% set skips = skips ~ " or (TestCustomOp and test_functionalize_error)" %}     # [py==313]
+        {% set skips = skips ~ " or (TestCustomOpAPI and test_compile)" %}              # [py==313]
+        {% set skips = skips ~ " or (TestCustomOpAPI and test_fake)" %}                 # [py==313]
```

Comment:
@mgorny, it seems that the failures match the tests here? So then it's just the selector that needs to be expanded

Reply:
Although it's strange - the skip for `TestTorch::test_print` is unconditional; I'm on a phone right now so cannot debug further, but very weird how the skips don't take effect.

Reply:
At a first glance, it seems that these are passed to pytest correctly, so my guess would be that `::` syntax doesn't work there. I'm going to try experimenting with it a little more in a while, lemme just tidy up some other stuff first.

Reply:
Confirmed that neither `::` nor `.` works — I guess you can't match both class name and method name via `-k`. Good news is that with the subset of tests we're running, it should be fine to just match on name, so I'll do that. I'll also do that for aarch64 skips, since I suspect they didn't really work as well.

Reply:
Oh, sorry, I've just realized I didn't pull the newest changes yet.

Reply:
Nevermind these, after pulling all tests pass for me. Perhaps https://github.com/conda-forge/pytorch-cpu-feedstock/actions/runs/12448456323/job/34771658661?pr=298 was a fluke?